client/my-sites/backup/backup-date-picker/test/index.jsx

```diff
@@ -56,205 +60,161 @@ describe( 'BackupDatePicker', () => {
+		const onDateChange = jest.fn();
+
+		render( <BackupDatePicker selectedDate={ selectedDate } onDateChange={ onDateChange } /> );
+
+		const button = screen.getByText( 'Dec 31, 2019' );
+		await user.click( button );
+
+		expect( onDateChange ).toHaveBeenCalledWith( selectedDate.subtract( 1, 'day' ) );
+	} );
+
+	test( 'Navigates forward when the next date is clicked', async () => {
+		const user = userEvent.setup();
+
+		const selectedDate = moment( '2020-01-01' );
+		const onDateChange = jest.fn();
+
+		render( <BackupDatePicker selectedDate={ selectedDate } onDateChange={ onDateChange } /> );
+
+		const button = screen.getByText( 'Jan 2, 2020' );
+		await user.click( button );
+
+		expect( onDateChange ).toHaveBeenCalledWith( selectedDate.add( 1, 'day' ) );
+	} );
+
+	test( 'Navigates backward when the previous date is focused and the spacebar is pressed', async () => {
+		const user = userEvent.setup();
+
+		const selectedDate = moment( '2020-01-01' );
+		const onDateChange = jest.fn();
+
+		render( <BackupDatePicker selectedDate={ selectedDate } onDateChange={ onDateChange } /> );
+
+		const button = screen.getByText( 'Dec 31, 2019' );
+		button.focus();
+		await user.keyboard( '[Space]' );
+
+		expect( onDateChange ).toHaveBeenCalledWith( selectedDate.subtract( 1, 'day' ) );
+	} );
+
+	test( 'Navigates forward when the next date is focused and the spacebar is pressed', async () => {
+		const user = userEvent.setup();
+
+		const selectedDate = moment( '2020-01-01' );
+		const onDateChange = jest.fn();
+
+		render( <BackupDatePicker selectedDate={ selectedDate } onDateChange={ onDateChange } /> );
+
+		const button = screen.getByText( 'Jan 2, 2020' );
+		button.focus();
+		await user.keyboard( '[Space]' );
+
+		expect( onDateChange ).toHaveBeenCalledWith( selectedDate.add( 1, 'day' ) );
+	} );
 
 	// --- ANALYTICS ---
 
-	test( 'Records a Tracks event for backward navigation', () =>
-		new Promise( ( done ) => {
-			const checkPreviousDateTracksEvent = ( event ) => {
-				const name = getTracksEventName( event );
-				expect( name ).toEqual( 'calypso_jetpack_backup_date_previous' );
-				done();
-			};
+	test( 'Records a Tracks event for backward navigation', async () => {
+		const user = userEvent.setup();
+		const selectedDate = moment( '2020-01-01' );
 
-			useDispatch.mockImplementation( () => checkPreviousDateTracksEvent );
+		const dispatchMock = jest.fn();
+		useDispatch.mockImplementation( () => dispatchMock );
 
-			const picker = shallow(
-				<BackupDatePicker selectedDate={ moment() } onDateChange={ () => {} } />
-			);
+		render( <BackupDatePicker selectedDate={ selectedDate } onDateChange={ () => {} } /> );
 
-			picker.find( '.backup-date-picker__select-date--previous' ).simulate( 'click' );
-		} ) );
+		await user.click( screen.getByText( 'Dec 31, 2019' ) );
+		expect( dispatchMock.mock.calls[ 0 ][ 0 ].meta.analytics[ 0 ].payload.name ).toEqual(
+			'calypso_jetpack_backup_date_previous'
+		);
```

Comment:
Instead of directly accessing `.mock.calls`, let's use a more expressive and idiomatic way:

```suggestion
		expect( dispatchMock ).toHaveBeenCalledWith(
			expect.objectContaining( {
				meta: {
					analytics: [
						expect.objectContaining( {
							payload: expect.objectContaining( {
								name: 'calypso_jetpack_backup_date_previous',
							} ),
						} ),
					],
				},
			} )
		);
```